Scripts/Data/Type/Transformation/Vector2ToAngle.cs

```diff
@@ -0,0 +1,107 @@
+﻿namespace Zinnia.Data.Type.Transformation
+{
+    using UnityEngine;
+    using UnityEngine.Events;
+    using System;
+    using Zinnia.Extension;
+
+    /// <summary>
+    /// Transforms a <see cref="Vector2"/> value to a <see cref="float"/> angle.
+    /// </summary>
+    /// <remarks>
+    /// The origin angle (i.e. zero) is set to the top by default to mimic a clock with 12 o'clock being the 0 angle.
+    /// </remarks>
+    /// <example>
+    /// Vector2[0f, 0f] -> unit.degrees -> origin.Vector2(0f, 1f) = 0f
+    /// Vector2[1f, 0f] -> unit.degrees -> origin.Vector2(0f, 1f) = 90f
+    /// Vector2[0f, -1f] -> unit.degrees -> origin.Vector2(0f, 1f) = 180f
+    /// </example>
+    public class Vector2ToAngle : Transformer<Vector2, float, Vector2ToAngle.UnityEvent>
+    {
+        /// <summary>
+        /// Defines the event with the transformed <see cref="float"/> value.
+        /// </summary>
+        [Serializable]
+        public class UnityEvent : UnityEvent<float>
+        {
+        }
+
+        public enum AngleUnit
+        {
+            /// <summary>
+            /// The angle in degrees.
+            /// </summary>
+            Degrees,
+            /// <summary>
+            /// The angle in radians.
+            /// </summary>
+            Radians
+        }
+
+        /// <summary>
+        /// The unit to return the converted angle as.
+        /// </summary>
+        [Tooltip("The unit to return the converted angle as.")]
+        public AngleUnit unit = AngleUnit.Degrees;
+
+        [Tooltip("The direction that defines the origin (i.e. zero) angle."), SerializeField]
+        private Vector2 _origin = new Vector2(0f, 1f);
+        /// <summary>
+        /// The direction that defines the origin (i.e. zero) angle..
+        /// </summary>
+        public Vector2 Origin
+        {
+            get { return _origin; }
+            set
+            {
+                _origin = value;
+                NormalizeOrigin();
+            }
+        }
+
+        /// <summary>
+        /// The full circle.
+        /// </summary>
+        protected const double fullCircle = 2f * Math.PI;
+
+        /// <summary>
+        /// Transforms the given input <see cref="Vector2"/> to the angle.
+        /// </summary>
+        /// <param name="input">The value to transform.</param>
+        /// <returns>The transformed value.</returns>
+        protected override float Process(Vector2 input)
+        {
+            if (input.sqrMagnitude.ApproxEquals(0f))
+            {
+                return 0f;
+            }
+
+            return (float)
+                ((Math.Atan2(Origin.y, Origin.x) - Math.Atan2(input.y, input.x) + fullCircle) % fullCircle
+                    * (unit == AngleUnit.Degrees ? 180f / Math.PI : 1f));
+        }
+
+        protected virtual void OnEnable()
```

Comment:
With an implementation of `OnValidate` (once the `if (!Application.isPlaying)` in there is removed) doing the same this is unnecessary:
The value can't be changed without going through `OnValidate` in the Editor and at runtime a user can only go through the property's setter.